tools/ui-components/src/close-button/close-button.tsx

```diff
@@ -15,18 +15,41 @@ export function CloseButton({
   onClick
 }: CloseButtonProps): JSX.Element {
   const classes = [
-    'text-xl font-bold leading-none appearance-none opacity-20',
-    'hover:opacity-50 focus:opacity-50',
+    // Remove browser's default styles
+    'bg-transparent',
+    'border-none',
+    // Text styles
+    'text-lg',
+    'font-bold',
+    'text-foreground-primary',
+    // Active state
+    'active:before:w-full',
+    'active:before:h-full',
+    'active:before:absolute',
+    'active:before:inset-0',
+    'active:before:border-3',
+    'active:before:border-transparent',
+    'active:before:bg-gray-900',
+    'active:before:opacity-20',
+    // Focus state
+    'focus:outline-none', // Hide the default browser outline
+    'focus:ring',
+    'focus:ring-focus-outline-color',
```

Comment:
We are using `:focus-visible` in `/learn` for these buttons, I will update the button component in https://github.com/freeCodeCamp/freeCodeCamp/pull/51443
```suggestion
    'focus-visible:ring',
    'focus-visible:ring-focus-outline-color',
```